core/src/dird/getmsg.cc

```diff
@@ -83,6 +83,12 @@ static void SetJcrSdJobStatus(JobControlRecord* jcr, int SDJobStatus)
   }
   jcr->dir_impl->SDJobStatus = SDJobStatus;
 
+  jcr->setJobStatus(SDJobStatus);
```

Comment:
TBH I'm not sure how you would replicate the SDJobStatus (and also the FDJobStatus) into the Director's job status correctly.
Just copying it -- like you did here -- may or may not be the right way.